### PR TITLE
Implement BS.2076-2 AudioChannelFormatIDRef in AudioTrackUID

### LINKED ISSUE #70: Add audioChannelFormatIDRef sub-element to audioTrackUID
In [2076-2](https://www.itu.int/dms_pubrec/itu-r/rec/bs/R-REC-BS.2076-2-201910-I!!PDF-E.pdf), audioChannelFormatIDRef sub-element was added to audioTrackUID (enabling bypassing of audioTrackFormat and audioStreamFormat for PCM only).

>Reference to an audioChannelFormat description.  This  element  is used  only  if  an audioTrackFormat  is  omitted  for  PCM  audio. Then  the  ‘yyyyxxxx’  parts  of  AC_yyyyxxxx and AT_yyyyxxxx_zz are the same number.

### Definition of done
* Able to parse an adm XML document containing an audioTrackUID with an audioChannelFormat reference, and write it back without modification
* Able to create a document with and audioTrackUID referencing an audioChannelFormat and write that to xml
* Have investigated the effort required to update libbw64 to support chna chunks containing an audioChannelFormat
* Provide user feedback in cases where both audioTrackFormat and audioChannelFormat references are present in a parsed xml (it should not be possible to write such a document using the library)

### Stretch goal
* Update libbw64 to support chna chunks containing audioChannelFormat reference (if necessary)

### PR DESCRIPTION
**Pull request by kristianhentschelbbc**
Closes #70.

### CODE REVIEW

**Review by tomjnixon — COMMENTED** [review]
This looks good to me!

Might it be good to update the route tracer? I'd be tempted to add the new reference but disable it with a `shouldRecurse` specialisation, so that the output from that doesn't change, but it's easy to use the new route without having to subclass the route tracer itself.

There are a few minor formatting things which might be worth fixing (indents in xml files, 

It might be worth checking the formatting before merging? e.g. there's a missing newline at the end of tests/xml_parser_audio_track_uid_tests.cpp, and the xml files look like they have some odd indents.

I guess it should be squashed before merging, as this looks like one change.

File: src/document.cpp [hunk 1]
```diff
@@ -89,30 +89,28 @@ namespace adm {
...
-  bool Document::add(
-          std::shared_ptr<AudioPackFormat> packFormat
-          ) {
-      if (packFormat->getParent().lock() &&
-          packFormat->getParent().lock() != shared_from_this()) {
-        throw std::runtime_error(
-            "AudioPackFormat already belongs to another Document");
+  bool Document::add(std::shared_ptr<AudioPackFormat> packFormat) {
+    if (packFormat->getParent().lock() &&
+        packFormat->getParent().lock() != shared_from_this()) {
```
[the remark this answers is not in the document]
Comment (kristianhentschelbbc) [reply src/document.cpp:94]: Sorry about this diff - there were a few instances of clang-format fixing something elsewhere in a file. (If we don't already, should we run the formatter in CI?)

[the remark this answers is not in the document]
Comment (tomjnixon) [reply src/document.cpp:94]: Yeah, that would be useful.

**Comment by kristianhentschelbbc** [comment]
Thanks for the review!

I've updated the route tracer with a `shouldRecurse` specialisation for the new reference. Is this what you meant (I'm not sure where else it is used and whether these changes are sufficient)?

> Might it be good to update the route tracer? I'd be tempted to add the new reference but disable it with a `shouldRecurse` specialisation, so that the output from that doesn't change, but it's easy to use the new route without having to subclass the route tracer itself.

**Comment by tomjnixon** [comment]
Looks great, thanks!

**Review by tomjnixon — APPROVED** [review]

**Pull request merged**